backend/errors/error_source.go

```diff
@@ -0,0 +1,37 @@
+package errors
```

Comment:
Would highly encourage not to have our own `errors` package as it just causes confusion and frequent collision with the standard Go library. If we do need this, then I'd just put it in its own file within the `backend` package.

Reply:
Was on my way to comment on that as well also related to we might end up using https://github.com/grafana/gerror/ eventually/soonish. Given the added middleware I suspect there might be a problem with import cycles putting this in the backend package, but not sure. What about naming this package `errorsource` for now?

Reply:
@marefr you are right. having it in backend was causing a circular reference between backend and httpclient

Reply:
removed